client/hawc_client/__init__.py

```diff
@@ -0,0 +1,306 @@
+from typing import Dict, Generator, List
+
+import pandas as pd
+from requests import Response, Session
+
+
+class HawcClientException(Exception):
+    """An exception occurred in the HAWC client module."""
+
+    pass
+
+
+class HawcServerException(Exception):
+    """An exception occurred on the HAWC server."""
+
+    pass
+
+
+class _Session:
+    """
+    A session that handles HAWC requests and responses.
+
+    Allows user authentication and keeps track of root url.
+    """
+
+    def __init__(self, root_url: str = "https://hawcproject.org"):
+        self.root_url = root_url
+        self._session = Session()
+
+    def _handle_hawc_response(self, response: Response) -> Dict:
+        """
+        Handle standard hawc API response
+
+        Args:
+            response (Response): A response object
+
+        Raises:
+            HawcClientException: If error occurs in data submission
+            HawcServerException: If error orccurs on server
+
+        Returns:
+            Dict: The JSON response, if exists
+        """
+        if response.status_code >= 400 and response.status_code < 500:
+            raise HawcClientException(response.status_code, response.json())
+        elif response.status_code == 500:
+            raise HawcServerException(response)
+        else:
+            return response.json()
+
+    def _get(self, url: str, params: Dict = None) -> Dict:
```

Comment:
since we refactored this class, these methods are now called by other external classes, so these should be public not private; change `_get` to `get`; same with other methods